lib/linter-reek.js

```diff
@@ -52,21 +52,12 @@ const parseJSON = (stdout) => {
 
 export default {
   activate() {
-    this.idleCallbacks = new Set();
-    let depsCallbackID;
-    const installLinterReekDeps = () => {
-      this.idleCallbacks.delete(depsCallbackID);
-      if (!atom.inSpecMode()) {
-        require('atom-package-deps').install('linter-reek');
-      }
-      loadDeps();
-      if (atom.inDevMode()) {
-        // eslint-disable-next-line no-console
-        console.log('linter-reek: All dependencies installed.');
-      }
-    };
-    depsCallbackID = window.requestIdleCallback(installLinterReekDeps);
-    this.idleCallbacks.add(depsCallbackID);
+    require('atom-package-deps').install('linter-reek', true);
+    loadDeps();
```

Comment:
Just did some local testing on this, and it looks like the file itself (`linter-reek.js`) is only loaded when the activation hook is triggered... but it **blocks** the opening of the file until the package is done calling `activate()`.

This means that all the optimizations we were doing before to reduce the load during Atom's initial startup are still useful for packages that use `activationHooks` because we don't want to be blocking the UI any more than we must.

To put it another way:
Move all of this back into the `idleCallback`. 😛